includes/class-wc-gateway-stripe.php

```diff
@@ -456,6 +456,10 @@ public function process_payment( $order_id, $retry = true, $force_save_source =
 				if ( 'requires_action' === $intent->status ) {
 					$this->unlock_order_payment( $order );
 
+					// If the order requires some action from the customer, add meta to the order to prevent it from being cancelled by WooCommerce's hold stock settings.
```

Comment:
Can we move this to a new helper method? Since it is being used twice

Reply:
I added helper methods in ab133cff3eca9ebe15725079e71d5977b939fead. Let me know your thoughts and thanks for the feedback!

Reply:
Looking great! Thank you